### PR TITLE
fix: fixed 1st deploy race condition when generating site from a template

### LINKED ISSUE #4212: Potential race condition in templates feature
**Describe the bug**

After the release of the templates command `sites:create-template` we ran into what seems to be a race condition issue that causes the 1st deploy of a new site to occasionally fail with the error `Failed during stage 'preparing repo': git ref refs/heads/main does not exist`.

It seems like even though the new repo is created on GitHub from a template before creating the new site on Netlify, when the deploy starts, it sometimes can't find the main branch?! 

Re-deploying works perfectly fine but we were wondering if there would be a fix for this to ensure this issue doesn't happen. We considered adding some kind of `setTimeout` before calling the Netlify API to create a new site but it feels a little weird 😕 and it might not even ensure that it works if one day GitHub is just very slow. 

We considered asking the team if we could implement some changes to re-try deploys automatically when this error happens but that would be outside of the scope of the CLI.

If you have any other ideas, please let us know!

**To Reproduce**

Steps to reproduce the behavior (it only occasionally happen so you might have to try a few times):

1. Run `netlify sites:create-template`
2. Pick the Gatsby or Hugo template
3. Finish running the command
4. Sometimes, the 1st deploy on Netlify will automatically fail with the error mentioned above.

**Expected behavior**

Creating a site from a template via the CLI should succeed on 1st deploy.

--

cc @maxcell & @tzmanics

### PR DESCRIPTION
**Pull request by nickytonline**
🎉 Thanks for submitting a pull request! 🎉

## Summary

Fixes #4212

As the issue explains, there is a race condition that occasionally occurs on the first deployment of a site when using the CLI command `ntl sites:create-template`.

Currently the CLI clones a repo using GitHub's API via the template endpoint.

https://github.com/netlify/cli/blob/7ffcbdfbdb5ccbc7e82a3550e46412b380e7d9e6/src/utils/sites/utils.js#L34-L47

The repository is created and the proper information is returned, but sometimes the refs aren't available right away when the repository is cloned when the site is being created.

Doing an explicit call to get the repository information after it's been created appears to fix the issue. My assumption is the repo is fully formed at this point, i.e. no `Failed during stage 'preparing repo': git ref refs/heads/main does not exist` error.

## QA Instructions, Screenshots, Recordings

1. Pull down this PR, e.g. via GitHub CLI `gh co 4607`
2. Follow the https://github.com/netlify/cli#development steps to get set up to test the PR
3. run `ntl sites:create-template` from the shell
4. Select any template. It doesn't matter which one.
5. Answer no for the step about cloning the repository.
6. Repeat steps 1-5 a bunch of times.
7. Head to your overview page, e.g. https://app.netlify.com/teams/your-team/overview, to see if any of your initial deployments fail. They shouldn't 🤞🏻 

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

Mike Diva playing the saxophone

### CODE REVIEW

**Comment by nickytonline** [comment]
Not sure if this is a known issue or it's just my PR, but he CLI fails to install for Mac

```
  ✔ yarn → should install the cli and run the help command (1m 5.2s)
  ─

  pnpm → should install the cli and run the help command

  Rejected promise returned by test. Reason:

  Error {
    command: 'pnpm add http://localhost:8840/netlify-cli/-/netlify-cli-10.3.1.tgz',
    escapedCommand: 'pnpm add "http://localhost:8840/netlify-cli/-/netlify-cli-10.3.1.tgz"',
    exitCode: 1,
    failed: true,
    isCanceled: false,
    killed: false,
    shortMessage: 'Command failed with exit code 1: pnpm add http://localhost:8840/netlify-cli/-/netlify-cli-10.3.1.tgz',
    signal: undefined,
    signalDescription: undefined,
    stderr: undefined,
    stdout: undefined,
    timedOut: false,
    message: 'Command failed with exit code 1: pnpm add http://localhost:8840/netlify-cli/-/netlify-cli-10.3.1.tgz',
  }

  › makeError (node_modules/execa/lib/error.js:60:11)
  › handlePromise (node_modules/execa/index.js:118:26)
  › async file://e2e/install.e2e.mjs:31:5

  ─

  1 test failed
Command failed with exit code 1: ava **/*.e2e.mjs --config /Users/runner/work/cli/cli/e2e.config.mjs
Error: Process completed with exit code 1.
```

**Review by tzmanics — DISMISSED** [review]
Thank you for working on this. Works great 👍

File: src/commands/sites/sites-create-template.js [hunk 1]
```diff
@@ -135,14 +135,16 @@ const sitesCreateTemplate = async (repository, options, command) => {
           )
         }
       } else {
+        const { default_branch: branch, full_name: repo, private } = await getRepo(repoResp.full_name, ghToken)
```
Comment (MarcL) [inline src/commands/sites/sites-create-template.js:138]: Could this potentially still fail or does it need a retry?
I'm just wondering whether we're guaranteed to have the repo created in a timely enough manner or whether we could get a 404 here.

Comment (MarcL) [reply src/commands/sites/sites-create-template.js:138]: That makes sense @nickytonline - thanks for the explanation.

**Pull request closed without merging**